### PR TITLE
Test environment

### PR DESCRIPTION
**Pull request by michel2323**
Removing JuMP from the dependencies and creating a test environment the "new" way without `[extras]` and `[targets]`.

### CODE REVIEW

File: Project.toml [hunk 1]
```diff
@@ -10,7 +10,6 @@ ExaTron = "28b18bf8-76f9-41ea-81fa-0f922810b349"
 FileIO = "5789e2e9-d7fb-5bc7-8068-2c6fae9b9549"
 Ipopt = "b6b21f68-93f8-5de0-b562-5493be1d77c9"
 JLD2 = "033835bb-8acc-5ee8-8aae-3f567f8a3819"
```
Comment (frapac) [inline Project.toml:12]: Do we depend on JLD2? This is large dep, and it would be nice to remove it if we can

File: test/Artifacts.toml [hunk 2]
```diff
@@ -0,0 +1 @@
+../Artifacts.toml
```
Comment (frapac) [inline test/Artifacts.toml:1]: Not sure about this one ... is it recommended in the Julia doc? If not I would prefer to copy/paste the Artifacts.toml in the test folder

File: test/Project.toml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+[deps]
+CUDA = "052768ef-5323-5732-b1bb-66c8b64840ba"
+JuMP = "4076af6c-e467-56ae-b986-b466b2749572"
```
Comment (kibaekkim) [inline test/Project.toml:3]: We can remove `JuMP` from test too.

File: Project.toml [hunk 4]
```diff
@@ -10,7 +10,6 @@ ExaTron = "28b18bf8-76f9-41ea-81fa-0f922810b349"
...
-JuMP = "4076af6c-e467-56ae-b986-b466b2749572"
 LazyArtifacts = "4af54fe1-eca0-43a8-85a7-787d91b784e3"
```
Comment (kibaekkim) [inline Project.toml:13]: It may be sufficient to have this for testing. So we can remove this too.

File: Project.toml [hunk 5]
```diff
@@ -26,5 +25,4 @@ FileIO = "1.14"
 Ipopt = "1"
```
Comment (kibaekkim) [inline Project.toml:25]: We can remove `Ipopt` too.

**Review by kibaekkim — COMMENTED** [review]
Please see my inline comments.

File: test/Artifacts.toml (hunk 2 above)
Comment (michel2323) [reply test/Artifacts.toml:1]: Why are links bad?

Comment (michel2323) [reply test/Artifacts.toml:1]: Okay, fixed in a clean way.

File: Project.toml (hunk 4 above)
Comment (michel2323) [reply Project.toml]: Done.

File: Project.toml (hunk 1 above)
Comment (michel2323) [reply Project.toml]: Removed.

**Review by kibaekkim — APPROVED** [review]
Looks good to me. I see the CPU test failed?

**Pull request merged**